pom.xml

```diff
@@ -42,6 +42,8 @@
   <url>http://blueflood.io</url>
 
   <properties>
+    <VERSION_SUFFIX>0-SNAPSHOT</VERSION_SUFFIX>
+    <BLUEFLOOD_VERSION>2.0.${VERSION_SUFFIX}</BLUEFLOOD_VERSION>
     <maven.javadoc.skip>true</maven.javadoc.skip>
```

Comment:
Sorry for the long message. I thought of talking to you but this way, others can weigh in.

I noticed an issue with the current approach. When using a property(BLUEFLOOD_VERSION) for 'version', it doesnt seem to play well with maven. It seems to generate the warning below when I tried to run 'mvn clean install'. I do not know the implications of the warning. I am using maven 3.3.3.

```
[WARNING] Some problems were encountered while building the effective model for com.rackspacecloud:blueflood-core:jar:2.0.0-SNAPSHOT
[WARNING] 'version' contains an expression but should be a constant. @ com.rackspacecloud:blueflood:${BLUEFLOOD_VERSION}, /Users/chan5120/Documents/workspacecf2/blueflood/pom.xml, line 26, column 14
```
Also this kinda version representation seems a little odd for a typical maven project.
```
<VERSION_SUFFIX>0-SNAPSHOT</VERSION_SUFFIX>
<BLUEFLOOD_VERSION>2.0.${VERSION_SUFFIX}</BLUEFLOOD_VERSION>
```

I was trying to avoid both and see if it is still possible to generate a release with a build number and I have the approach as outlined below. 

1) Lets leave the pom.xml's the way they are. This way mvn can be run directly without the build script.
2) We can use ``mvn versions:set`` to change versions of pom.xml in all the modules to have 2.0.BUILD_NUMBER. We will **NOT** commit the changed pom files. 
3) We do the ``mvn package -P all-modules`` to get released version of jar file(2.0.BUILD_NUMBER)
4) tag the release.

For STEP 2, we do have the option of creating a separate release branch with name '2.0.BUILD_NUMBER', commit the changed pom.xml in that branch for future reference, if we desire. Its upto us. 

We do have to change the version of pom.xml's dynamically. In your build script, you are doing it by passing a paramenter BUILD_NUMBER, and in the above approach, I am using ``mvn versions:set`` command to set version with BUILD_NUMBER.

CONs:
When we are ready for the next build. clean the directory to remove the files created by ``mvn versions:set`` from previous run and we repeat the steps. Can be done with something like ``git clean -f -d``. In Jenkins we can check 'Delete workspace before build starts' to address this.